scripts/09_deploy_openshift.sh

```diff
@@ -12,13 +12,39 @@ mkdir -p ocp/openshift
 python3 /root/bin/ipmi.py off
 python3 /root/bin/redfish.py off
 {% if bmc_reset %}
-  {% for worker in workers %}
-  {% if worker['model']|default('kvm') == "dell" %}
-      worker_ip={{ worker["redfish_address"] }}
-      curl -i -k -X POST -H 'Content-Type: application/json' -H 'Accept: application/json' -u {{ bmc_user }}:{{ bmc_password }} --data '{"ResetType":"GracefulRestart"}' 'https://"${worker_ip}"/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Manager.Reset'
-    {% endif %}
-  {% endfor %}
+{% for worker in workers %}
+{% if worker['model']|default('kvm') == "dell" %}
+worker_ip={{ worker["redfish_address"] }}
+curl -i -k -X POST -H "Content-Type: application/json" -H "Accept: application/json" -u root:calvin --data '{"ResetType":"GracefulRestart"}' "https://${worker_ip}/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Manager.Reset"
+if [[ $? -eq 0 ]]; then  # Can be implicit, but explicit for visibility
+  echo "BMC ${worker_ip} restarting!"
+  sleep 30  # Time to start restart
+else
+  echo "ERROR: BMC ${worker_ip} fails to restart"
+  # exit 1
+fi
+{% endif %}
+{% endfor %}
+{% endif %}
```

Comment:
is this "endif" of {% if bmc_reset %} ? If so, shouldn't be in the end of the whole code block of checking bmc?

Reply:
I don't think so, because checking that the BMCs are up can be considered as a necessary task for the Dell workers regardless of the reset. We can also add a check for HPE.

Reply:
OK